### DISCUSSION TITLE
Unable to save member

### DISCUSSION THREAD

**Discussion by yepadoos** [codefori/vscode-ibmi#1162] [General]
I am unable to save a member in Vscode.  I am getting the following error and messages in the Code for Ibmi console.

![image](https://user-images.githubusercontent.com/104451298/227325313-d4769fba-bcb3-43ea-9d17-6118cba5a3ee.png)

**Comment by worksofliam** [comment]
@yepadoos Is this every member or a specific one? More detail please.

* How many lines of code?
* OS version?
* SQL enabled/disabled?
* Source dates on/off?

**Reply by yepadoos** [reply]
Not every member just this specific one.  Enable SQL not checked, Enable Source Dates is checked.

**Reply by yepadoos** [reply]
Lines of code 3027, OS V7R3M0

**Reply by worksofliam** [reply]
1.7.8 is out which will solve this problem!

**Comment by worksofliam** [comment]
@chrjorgensen I wonder if the addition of the code that does the hex conversion has had an impact on large sources because it has made our SQL longer.

**Reply by worksofliam** [reply]
<img width="508" alt="image" src="https://user-images.githubusercontent.com/3708366/227337760-7d37eae6-1ea3-484e-b6b9-a81868012def.png">

**Reply by worksofliam** [reply]
Like of code here: https://github.com/halcyon-tech/vscode-ibmi/blob/807050034aa9e552360363a4725529e7e7a445aa/src/filesystems/qsys/extendedContent.ts#L132

**Reply by chrjorgensen** [reply]
@worksofliam You're error seems to come from the `readonly` function implemented in PR #1075?
Or is it just the error message, that includes the `readonly` reference?

**Reply by worksofliam** [reply]
It's coming from the just below the line of code I linked to. I am debugging and it seems the insert query is too large for `insert into ...`, which is what the `sliceUp` fixed previously, but didn't account for those extra translate characters!

edit: pr incoming

**Reply by chrjorgensen** [reply]
I get this error when I try to open a member without any rows:

![billede](https://user-images.githubusercontent.com/13275072/227345079-b719418f-85a1-498e-8369-599e16ccd2f3.png)

**Reply by worksofliam** [reply]
@chrjorgensen I also fixed that bug in my PR.

**Reply by chrjorgensen** [reply]
Yeah, saw that - after my error report... 😆
